### ISSUE TITLE
1.7 - Clicking on a result doesn't reveal it correctly

### ISSUE THREAD

**Issue by Miraihi** [Acly/krita-ai-diffusion#52]
The new bug has been introduced in 1.7. Usually if click on a result in the plugin's menu, it toggles visibility of the preview on and off. In update 1.7 it only hides the result but doesn't reveal it, requiring me to interact with the layers menu. If I try to add the generated image via doubleclicking the result as usual, I get the error:

```
AssertionError
Python 3.10.7: C:\Program Files (x86)\Steam\steamapps\common\Krita\krita\bin\krita.exe
Fri Nov 17 16:49:56 2023

A problem occurred in a Python script.  Here is the sequence of
function calls leading up to the error, in the order they occurred.

 C:\Users\Miraihi\AppData\Roaming\krita\pykrita\ai_diffusion\ui\widget.py in apply_result(self=<ai_diffusion.ui.widget.GenerationWidget object>, item=<PyQt5.QtWidgets.QListWidgetItem object>)
  753     def apply_result(self, item: QListWidgetItem):
  754         self.show_preview(item)
  755         self.apply_selected_result()
  756 
  757 
self = <ai_diffusion.ui.widget.GenerationWidget object>
self.apply_selected_result = <bound method GenerationWidget.apply_selected_re...<ai_diffusion.ui.widget.GenerationWidget object>>

 C:\Users\Miraihi\AppData\Roaming\krita\pykrita\ai_diffusion\ui\widget.py in apply_selected_result(self=<ai_diffusion.ui.widget.GenerationWidget object>)
  749 
  750     def apply_selected_result(self):
  751         self.model.apply_current_result()
  752 
  753     def apply_result(self, item: QListWidgetItem):
self = <ai_diffusion.ui.widget.GenerationWidget object>
self.model = <ai_diffusion.ui.model.Model object>
self.model.apply_current_result = <bound method Model.apply_current_result of <ai_diffusion.ui.model.Model object>>

 C:\Users\Miraihi\AppData\Roaming\krita\pykrita\ai_diffusion\ui\model.py in apply_current_result(self=<ai_diffusion.ui.model.Model object>)
  415     def apply_current_result(self):
  416         """Promote the preview layer to a user layer."""
  417         assert self._layer and self.can_apply_result
  418         self._layer.setLocked(False)
  419         self._layer.setName(self._layer.name().replace("[Preview]", "[Generated]"))
self = <ai_diffusion.ui.model.Model object>
self._layer = <PyKrita.krita.Node object>
self.can_apply_result = False
AssertionError: 
    __cause__ = None
    __class__ = <class 'AssertionError'>
    __context__ = None
    __delattr__ = <method-wrapper '__delattr__' of AssertionError object>
    __dict__ = {}
    __dir__ = <built-in method __dir__ of AssertionError object>
    __doc__ = 'Assertion failed.'
    __eq__ = <method-wrapper '__eq__' of AssertionError object>
    __format__ = <built-in method __format__ of AssertionError object>
    __ge__ = <method-wrapper '__ge__' of AssertionError object>
    __getattribute__ = <method-wrapper '__getattribute__' of AssertionError object>
    __gt__ = <method-wrapper '__gt__' of AssertionError object>
    __hash__ = <method-wrapper '__hash__' of AssertionError object>
    __init__ = <method-wrapper '__init__' of AssertionError object>
    __init_subclass__ = <built-in method __init_subclass__ of type object>
    __le__ = <method-wrapper '__le__' of AssertionError object>
    __lt__ = <method-wrapper '__lt__' of AssertionError object>
    __ne__ = <method-wrapper '__ne__' of AssertionError object>
    __new__ = <built-in method __new__ of type object>
    __reduce__ = <built-in method __reduce__ of AssertionError object>
    __reduce_ex__ = <built-in method __reduce_ex__ of AssertionError object>
    __repr__ = <method-wrapper '__repr__' of AssertionError object>
    __setattr__ = <method-wrapper '__setattr__' of AssertionError object>
    __setstate__ = <built-in method __setstate__ of AssertionError object>
    __sizeof__ = <built-in method __sizeof__ of AssertionError object>
    __str__ = <method-wrapper '__str__' of AssertionError object>
    __subclasshook__ = <built-in method __subclasshook__ of type object>
    __suppress_context__ = False
    __traceback__ = <traceback object>
    args = ()
    with_traceback = <built-in method with_traceback of AssertionError object>

The above is a description of an error in a Python program.  Here is
the original traceback:

Traceback (most recent call last):
  File "C:\Users\Miraihi\AppData\Roaming\krita\pykrita\ai_diffusion\ui\widget.py", line 755, in apply_result
    self.apply_selected_result()
  File "C:\Users\Miraihi\AppData\Roaming\krita\pykrita\ai_diffusion\ui\widget.py", line 751, in apply_selected_result
    self.model.apply_current_result()
  File "C:\Users\Miraihi\AppData\Roaming\krita\pykrita\ai_diffusion\ui\model.py", line 417, in apply_current_result
    assert self._layer and self.can_apply_result
AssertionError
```

**Comment by Miraihi** [comment]
[Fixed in 1.7.1](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.7.1)

**Comment by Miraihi** [comment]
[Fixed in 1.7.1](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.7.1)
